### PR TITLE
Add an example that helps use custom commands as "shell"

### LINKED ISSUE #2646: Add an example for custom shells
The documentation on [custom shells](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#custom-shell), unlike the rest of the shells, some of which are pretty obvious, does not include a good example with a real use case.

### What article on docs.github.com is affected?

Article on [workflow syntax](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions).

### What part(s) of the article would you like to see updated?

As indicated above, the sub section on "Custom shells"

### Additional information

I found that as a solution to [this problem I had](https://github.community/t/trying-to-create-perl-actions-and-finding-some-errors/154131/2), where I wanted to use perl as a shell.

### PR DESCRIPTION
**Pull request by JJ**
And, at the same time, helps people find "perl" as a legit thing in Github Actions.

### Why:

#2646 

### What's being changed:

An example and intro to it is being added.

### Check off the following:
- [x] I have reviewed my changes in staging. (look for the **deploy-to-heroku** link in your pull request, then click **View deployment**)
- [x] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [x] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### CODE REVIEW

**Comment by janiceilene** [comment]
@JJ Thanks so much for opening a PR! I'll get this triaged for review ⚡

**Comment by JJ** [comment]
Ping?

File: content/actions/reference/workflow-syntax-for-github-actions.md [hunk 1]
```diff
@@ -688,7 +688,17 @@ steps:
 
 #### Custom shell
 
-You can set the `shell` value to a template string using `command […options] {0} [..more_options]`. {% data variables.product.prodname_dotcom %} interprets the first whitespace-delimited word of the string as the command, and inserts the file name for the temporary script at `{0}`.
+You can set the `shell` value to a template string using `command […options] {0} [..more_options]`. {% data variables.product.prodname_dotcom %} interprets the first whitespace-delimited word of the string as the command, and inserts the file name for the temporary script at `{0}`, as in this example:
+
```
Comment (skedwards88) [inline content/actions/reference/workflow-syntax-for-github-actions.md:692]: optional nit on spacing/structure:
```suggestion
You can set the `shell` value to a template string using `command […options] {0} [..more_options]`. {% data variables.product.prodname_dotcom %} interprets the first whitespace-delimited word of the string as the command, and inserts the file name for the temporary script at `{0}`.

For example:

```

File: content/actions/reference/workflow-syntax-for-github-actions.md [hunk 2]
```diff
@@ -688,7 +688,17 @@ steps:
...
+```yaml
+steps:
+  - name: Display the environment variables and their values
+    run: |
+      print %ENV
+    shell: perl {0}
+```
+
+The command used, `perl` in this case, needs to be installed in the environment, please check [the documentation on virtual environments](https://github.com/actions/virtual-environments) for the list of commands, languages and utilities available.
```
Comment (skedwards88) [inline content/actions/reference/workflow-syntax-for-github-actions.md:701]: I think this alternative link will give the user the info they need while still staying on the docs site (which can lead to a smoother user experience). I also made some small stylistic changes.

```suggestion
The command used, `perl` in this example, must be installed on the runner. For information about the software included on GitHub-hosted runners, see "[Specifications for GitHub-hosted runners](/actions/reference/specifications-for-github-hosted-runners#supported-software)."
```

**Review by skedwards88 — COMMENTED** [review]
🚀 Thank you for adding this example! I made a few suggestions around the wording. Let me know what you think!

**Review by skedwards88 — APPROVED** [review]
🚀 LGTM! I'll get this merged down for you. Thank you for your contribution!

**Pull request merged**